test/src/test/java/hudson/slaves/JNLPLauncherTest.java

```diff
@@ -31,36 +31,43 @@
 import hudson.model.Node;
 import hudson.model.Node.Mode;
 import hudson.model.Slave;
-import hudson.remoting.Callable;
 import hudson.remoting.Which;
 import hudson.util.ArgumentListBuilder;
 
 import jenkins.security.SlaveToMasterCallable;
-import org.jvnet.hudson.test.HudsonTestCase;
+
+import org.junit.Assume;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
 import org.jvnet.hudson.test.TestExtension;
 
-import javax.inject.Inject;
 import java.io.File;
 import java.net.URL;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.concurrent.TimeUnit;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
 import java.awt.*;
 
 /**
  * @author Kohsuke Kawaguchi
  */
-public class JNLPLauncherTest extends HudsonTestCase {
+public class JNLPLauncherTest {
+    @Rule public JenkinsRule j = new JenkinsRule();
+
     /**
      * Starts a JNLP slave agent and makes sure it successfully connects to Hudson. 
      */
+    @Test
     public void testLaunch() throws Exception {
-        if(GraphicsEnvironment.isHeadless()) {
-            System.err.println("Skipping JNLPLauncherTest.testLaunch because we are running headless");
-            return;
-        }
+        Assume.assumeFalse("Skipping JNLPLauncherTest.testLaunch because we are running headless", GraphicsEnvironment.isHeadless());
```

Comment:
no point in putting the test name as it is available in the test report ([example](https://jenkins.ci.cloudbees.com/job/core/job/jenkins-core/lastCompletedBuild/testReport/hudson/FunctionsTest/testGetRelativeLinkTo_JobNotContainedInView/)) and would be incorrect in any refactoring.